searchv2/rules.py

```diff
@@ -0,0 +1,405 @@
+from datetime import timedelta
+from django.utils import timezone
+from pydantic import BaseModel
+from typing import List
+from typing import Optional
+
+from package.models import Package
+
+
+class CheckResult(BaseModel):
+    """
+    Model to store the result of a check, which includes the score and a message.
+    """
+
+    score: int
+    message: str
+
+
+class ScoreRule(BaseModel):
+    """
+    Abstract base class for all score rules. It validates the input data using Pydantic,
+    and requires a `check` method that should be implemented by subclasses.
+
+    Attributes:
+        name: The name of the score rule.
+        description: The description of the score rule.
+        max_score: The maximum score that can be awarded by this rule.
+        documentation_url: An optional URL to the documentation for this rule.
+        # rules: An optional list of sub-rules.
+
+    Methods:
+        check: Checks the given package against this scoring rule and returns a score.
+    """
+
+    name: str
+    description: str
+    max_score: int
+    documentation_url: Optional[str] = None
+    # rules: Optional[List["ScoreRule"]] = None
+
+    class Config:
```

Comment:
This is deprecated on pydantic v2.x. Ref: https://docs.pydantic.dev/latest/migration/#changes-to-config